websocketpp/transport/asio/endpoint.hpp

```diff
@@ -518,17 +518,15 @@ class endpoint : public config::socket_type {
         lib::error_code & ec)
     {
         using lib::asio::ip::tcp;
-        tcp::resolver r(*m_io_service);
-        tcp::resolver::query query(host, service);
-        tcp::resolver::iterator endpoint_iterator = r.resolve(query);
-        tcp::resolver::iterator end;
-        if (endpoint_iterator == end) {
+        tcp::resolver r(*m_io_context);
+        tcp::resolver::results_type endpoints = r.resolve(host, service);
+        if (endpoints.empty()) {
             m_elog->write(log::elevel::library,
                 "asio::listen could not resolve the supplied host or service");
             ec = make_error_code(error::invalid_host_service);
             return;
         }
-        listen(*endpoint_iterator,ec);
+        listen(*endpoints.begin(),ec);
```

Comment:
Personal preference (for readability) is to use parenthesis when dereferencing like this so it is clear to which it is applying.  Something like this:
```suggestion
        listen(*(endpoints.begin()),ec);
```